src/WriteLockStrategy/MysqlMetadataLockStrategy.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * This file is part of prooph/pdo-event-store.
+ * (c) 2016-2019 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2019 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Pdo\WriteLockStrategy;
+
+use Prooph\EventStore\Pdo\WriteLockStrategy;
+
+final class MysqlMetadataLockStrategy implements WriteLockStrategy
```

Comment:
Not sure if I like the naming, metadata lock makes me think about event metadata. Why did you bring the term "metadata" here? Any reason behind it?

Reply:
Because `GET_LOCK` uses Mysql's Metadata Locking System. Hence the name specifiying the techonoly used for locking (e.g. in contrast to using a custom roled one like for projections, which still would have a look in its name)

Reply:
Everyday you can learn something new, thanks.